client/components/gravatar/index.jsx

```diff
@@ -46,22 +38,23 @@ export class Gravatar extends Component {
 		size: 32,
 	};
 
+	state = { failedToLoad: false };
+
 	getResizedImageURL( imageURL ) {
 		const { imgSize } = this.props;
 		imageURL = imageURL || 'https://www.gravatar.com/avatar/0';
-		const parsedURL = url.parse( imageURL );
-		const query = parse( parsedURL.query );
+		const parsedURL = getUrlParts( imageURL );
+		delete parsedURL.search;
 
 		if ( /^([-a-zA-Z0-9_]+\.)*(gravatar.com)$/.test( parsedURL.hostname ) ) {
-			query.s = imgSize;
-			query.d = 'mm';
+			parsedURL.searchParams.set( 's', imgSize );
+			parsedURL.searchParams.set( 'd', 'mm' );
 		} else {
 			// assume photon
-			query.resize = imgSize + ',' + imgSize;
+			parsedURL.searchParams.set( 'resize', `${ imgSize },${ imgSize }` );
 		}
 
-		parsedURL.search = stringify( query );
-		return url.format( parsedURL );
+		return getUrlFromParts( parsedURL ).href;
```

Comment:
well this reads much better to me than the previous code - I trust it more knowing it's being properly parsed and returned.